### ISSUE TITLE
DEADLINE_EXCEEDED after 60s

### ISSUE THREAD

**Issue by romanstingler** [stackkit/laravel-google-cloud-tasks-queue#48]
Hi, I have longer running tasks and after 60s it retries with the DEADLINE_EXCEEDED error

![image](https://user-images.githubusercontent.com/5576250/162175102-9c51661e-8b04-47f8-90fe-9adf4b5b6575.png)

I found the same issue here 
https://github.com/googleapis/python-tasks/issues/93

defining following doesn't work

```
        'cloudtasks' => [
            'driver' => 'cloudtasks',
            'project' => env('STACKKIT_CLOUD_TASKS_PROJECT', ''),
            'location' => env('STACKKIT_CLOUD_TASKS_LOCATION', ''),
            'handler' => env('STACKKIT_CLOUD_TASKS_HANDLER', ''),
            'queue' => env('STACKKIT_CLOUD_TASKS_QUEUE', 'default'),
            'service_account_email' => env('STACKKIT_CLOUD_TASKS_SERVICE_EMAIL', ''),
            'retry_after' => 3600,
            'dispatch_deadline' =>  (new \Google\Protobuf\Duration())->setSeconds(300),
        ],
```

I know that I put it to the connection, but also within the Job I  don't have any direct task handling functionality.

@marickvantuil Has anyone any idea how I could override this.

**Comment by marickvantuil** [comment]
Hey, I released `3.1.0` which allows setting the dispatch deadline through the config file:
<img width="870" alt="Screenshot 2022-04-09 at 15 58 46" src="https://user-images.githubusercontent.com/647007/162577484-18ab47ee-0be8-4f11-9744-d0b56680e8eb.png">

Please let me know if this fixes the issue!

**Comment by romanstingler** [comment]
> Hey, I released `3.1.0` which allows setting the dispatch deadline through the config file: <img alt="Screenshot 2022-04-09 at 15 58 46" width="870" src="https://user-images.githubusercontent.com/647007/162577484-18ab47ee-0be8-4f11-9744-d0b56680e8eb.png">
> 
> Please let me know if this fixes the issue!

Hi thanks a lot for the fast reply.

I tested it 
and I had to upgrade from 2.x to 3.1 
I removed the "/handle-task" from the URL 

The issue is that you concat 
` return $handler . '/handle-task';`
in
`src/Config.php:65`

this results that the handle URL is sent with "/handle-task" but this part is missing in the **.env** file which leads to
**Audience does not match** in
`src/AccessToken.php:271`


```
       if (property_exists($payload, 'aud')) {
            if ($audience && $payload->aud != $audience) {
                throw new UnexpectedValueException('Audience does not match');
            }
        }
```

The issue is that in 
`src/TaskHandler.php` the `$this->config` is missing the "/handle-task"

with this dirty line it works:
```
$this->config['handler']  = $this->config['handler'] . "/handle-task";
```

And also the 
`            'dispatch_deadline' =>  300,
`
parameter works as expected. So after this fix I consider the issue as resolved.

Thanks a lot

**Comment by marickvantuil** [comment]
Thanks! The 'audience does not match' bug is fixed in 3.1.1. Thanks for reporting.
